base/interactiveutil.jl

```diff
@@ -40,7 +40,7 @@ function edit(path::AbstractString, line::Integer=0)
     end
     background = true
     line_unsupported = false
-    if startswith(name, "emacs") || name == "gedit"
+    if startswith(name, "emacs") || name == "gedit" || startswith(name, "gvim")
```

Comment:
I am not a Linux expert, but as far as I understand emacs also can be run in the terminal. And secondly when one sets editor to gvim probably one wants to run it in GUI.